### DISCUSSION TITLE
Simpler dependency development via `"require-local": [ "dev/*/composer.json" ]`

### DISCUSSION THREAD

**Discussion by frodeborli** [composer/composer#9970] [General]
I would love it if we could have a simpler way to develop dependency libraries.

Something like:

```
{
    "require-local": [
        "vendor-dev/some-library", // path to where a composer.json file resides
        "vendor-dev/*", // glob support?
        "http://username@bitbucket.org....", // git support?
    ]
}
```

Whenever we run `composer dump-autoload`, composer would update the autoloader, and update any bin-files according to 'require-local'.

Any comments?

**Comment by Seldaek** [comment]
You can use path repositories: https://getcomposer.org/doc/05-repositories.md#path
Or studio https://github.com/franzliedke/studio for more advanced local dev use cases

And you can track https://github.com/composer/composer/issues/601 too which may bring this in composer itself one day..
